### ISSUE TITLE
How do we set up this repo?

### ISSUE THREAD

**Issue by JamesPHoughton** [Watts-Lab/deliberation-infrastructure#1]
There are a number of different services that we will want in the future to all be configurable in this repo:
1. deliberation-empirica
2. video coding tool
3. scheduling tool
4. researcher portal
5. CDN

Each of these will need some shared infrastructure (ie, share a vpc, or usesr/security group) but some of the resources will be just for a single service. How do we split these out?

From: https://developer.hashicorp.com/terraform/cloud-docs/recommended-practices/part1
> To delegate a large application, companies often split it into small, focused microservice components that are owned by specific teams. Each microservice provides an API, and as long as those APIs don't change, microservice teams can make changes in parallel despite relying on each others' functionality.
> 
> Similarly, infrastructure code can be split into smaller Terraform configurations, which have limited scope and are owned by specific teams. These independent configurations use [output variables](https://developer.hashicorp.com/terraform/language/values/outputs) to publish information and [remote state resources](https://developer.hashicorp.com/terraform/language/state/remote-state-data) to access output data from other workspaces. Just like microservices communicate and connect via APIs, Terraform workspaces connect via remote state.
> 
> Once you have loosely-coupled Terraform configurations, you can delegate their development and maintenance to different teams. To do this effectively, you need to control access to that code. Version control systems can regulate who can commit code, but since Terraform affects real infrastructure, you also need to regulate who can run the code.
> 
> This is how Terraform Cloud solves the organizational complexity of provisioning: by providing a centralized run environment for Terraform that supports and enforces your organization's access control decisions across all workspaces. This helps you delegate infrastructure ownership to enable parallel development.

**Comment by JamesPHoughton** [comment]
I think having a separate workspace for each service, plus one for the networking/shared infrastructure makes a lot of sense. That way, we can take down one of the services without it influencing the others. Having each of these workspaces correspond to a directory in this repo seems like a simple way to accomplish that. 

Having it set up this way means we'll need to use 'outputs' to let the various services know about each other. I havent done this yet, so we'll need to figure out how to do it. I have a fellow on upwork who can be helpful, if we want to set up a time to meet with him.
